manage_user_reset.php

```diff
@@ -52,33 +52,37 @@
 );
 
 $t_command = new UserResetPasswordCommand( $t_data );
-$t_command->execute();
+$t_result = $t_command->execute();
+$t_result = $t_result['result'];
 
 $t_redirect_url = 'manage_user_page.php';
 
 form_security_purge( 'manage_user_reset' );
 
-layout_page_header( null, $t_result ? $t_redirect_url : null );
-
+layout_page_header(
+	null,
+	$t_result != UserResetPasswordCommand::RESULT_FAILURE ? $t_redirect_url : null
+);
 layout_page_begin( 'manage_overview_page.php' );
 
-if( $t_reset ) {
-	if( false == $t_result ) {
-		# PROTECTED
-		html_operation_failure( $t_redirect_url, lang_get( 'account_reset_protected_msg' ) );
-	} else {
-		# SUCCESSFUL RESET
-		if( ( ON == config_get( 'send_reset_password' ) ) && ( ON == config_get( 'enable_email_notification' ) ) ) {
-			# send the new random password via email
+switch( $t_result ) {
+	case UserResetPasswordCommand::RESULT_RESET:
+		if(    ( ON == config_get( 'send_reset_password' ) )
```

Comment:
Why do we need to do config changes here?  We can key the language string from the action returned.

Reply:
Not sure what you mean by _config changes_ here.

> We can key the language string from the action returned.

Are you suggesting that the Command should return the language string to use ? If so it would require special handling in users_rest.php to remove this from the returned JSON, as I don't think this this is something that should be exposed to a client consuming the API.

Reply:
I meant why are we doing config checks here.  You can imagine that the command can return a email_sent = true or false, that is checked here and appropriate string is displayed to the user.  That is not blocking, we can do that as a follow up change as we separate the commands later.